packages/dashboards/addon/services/dashboard-data.js

```diff
@@ -36,54 +39,89 @@ export default Service.extend({
   /**
    * @method fetchDataForDashboard
    * @param {Object} dashboard - dashboard model
-   * @returns {Promise} - Promise that resolves to a dashboard object with widget data
+   * @returns {Promise} - Promise that resolves to a hash of widget id to data promise array
+   */
+  async fetchDataForDashboard(dashboard) {
+    const widgets = await dashboard.widgets;
+    const layout = get(dashboard, 'presentation.layout');
+
+    return this.fetchDataForWidgets(dashboard.id, widgets, layout, [], get(this, 'widgetOptions'));
+  },
+
+  /**
+   * @method cancelFetchDataForDashboard - cancels all running or enqueued fetch Task Instances
    */
-  fetchDataForDashboard(dashboard) {
-    return dashboard
-      .get('widgets')
-      .then(widgets => this.fetchDataForWidgets(dashboard.id, widgets, [], get(this, 'widgetOptions')));
+  cancelFetchDataForDashboard() {
+    this._fetchTask.cancelAll();
```

Comment:
Is it possible to have `fetchDataForWidgets` return a parent task that can be canceled. That way the singleton service doesn't have any _state_.